service/controller/v2/resource/instance/create.go

```diff
@@ -15,11 +17,94 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("EnsureCreated called for cluster ID '%s'", key.ClusterID(customObject)))
+	// Find the next instance ID and instance name we want to trigger the update
+	// for. Instance names look something like the following example.
+	//
+	//     0gjpt-worker-000004
+	//
+	// Instance IDs are simple non negative integers.
+	//
+	//     4
+	//
+	var instanceID string
+	var instanceName string
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the next instance to be updated")
 
-	// TODO list all instances
-	// TODO find the first instance not having the latest scale set model applied
-	// TODO trigger update for found instance
+		c, err := r.getVMsClient()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		g := key.ResourceGroupName(customObject)
+		s := key.WorkerVMSSName(customObject)
+		result, err := c.List(ctx, g, s, "", "", "")
+		if IsScaleSetNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the scale set")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, v := range result.Values() {
+			if key.IsFinalProvisioningState(*v.ProvisioningState) {
+				continue
+			}
+
+			instanceName = key.InstanceName(customObject, *v.InstanceID)
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("instance '%s' is in state '%s'", instanceName, *v.ProvisioningState))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		}
+
+		for _, v := range result.Values() {
+			if *v.LatestModelApplied {
+				continue
+			}
+
+			instanceID = *v.InstanceID
+			instanceName = key.InstanceName(customObject, *v.InstanceID)
+
+			break
+		}
+
+		if instanceID == "" {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "no instance ID found that needs to be updated")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found instance '%s' has to be updated", instanceName))
+	}
+
+	// Trigger the update for the found instance.
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring instance '%s' to be updated", instanceName))
```

Comment:
I think I told that already there is a `%q` directive for quotation. I think it's more idiomatic to format it like `fmt.Sprintf("ensuring instance %q to be updated", instanceName)`. This is a minor.

Reply:
I replied to you in the discussion above towards this.